src/widgets/latest-transactions/LatestTransactions.vue

```diff
@@ -62,54 +62,57 @@
 </template>
 
 <script setup lang="ts">
-import { computed, ref, shallowRef, watch } from 'vue';
+import { computed, type Reactive, reactive, watch } from 'vue';
 import TimeIcon from '@/shared/ui/icons/clock.svg';
-import type { filterTransactionsModel as ftm } from '@/features/filter-transactions';
 import { TransactionStatusFilter } from '@/features/filter-transactions';
 import TransactionStatus from '@/entities/transaction/TransactionStatus.vue';
 import BaseHash from '@/shared/ui/components/BaseHash.vue';
 import BaseButton from '@/shared/ui/components/BaseButton.vue';
 import BaseContentBlock from '@/shared/ui/components/BaseContentBlock.vue';
-import { defaultFormat } from '@/shared/lib/time';
 import BaseLoading from '@/shared/ui/components/BaseLoading.vue';
-import type { Transaction } from '@/shared/api/schemas';
-import { useErrorHandlers } from '@/shared/ui/composables/useErrorHandlers';
 import * as http from '@/shared/api';
 import { useWindowSize } from '@vueuse/core';
 import { LG_WINDOW_SIZE, XS_WINDOW_SIZE } from '@/shared/ui/consts';
-import { objectOmit } from '@vueuse/shared';
 import TimeStamp from '@/shared/ui/components/TimeStamp.vue';
+import { useParamScope } from '@vue-kakuyaku/core';
+import { handleParamScope } from '@/shared/api/handle-param-scope';
 
 const emit = defineEmits<{
   loaded: [number]
 }>();
 
-const status = ref<ftm.Status>(null);
-
-const transactions = shallowRef<Transaction[]>([]);
-const isLoading = ref(false);
-
-const { handleUnknownError } = useErrorHandlers();
+const listState = reactive({
+  per_page: 5,
+  status: null,
+});
 
-watch(
-  status,
-  async () => {
-    try {
-      isLoading.value = true;
+async function fetchTransactions(params: Reactive<typeof listState>) {
+  return await http.fetchTransactions({
+    ...params,
+    status: params.status ?? undefined,
+  });
+}
 
-      const params = { per_page: 5, status: status.value };
+const scope = useParamScope(
+  () => {
+    return {
+      key: Object.values(listState).join('-'),
+      payload: listState,
+    };
+  },
+  ({ payload }) => handleParamScope(payload, fetchTransactions)
+);
 
-      const res = await http.fetchTransactions(objectOmit(params, status.value ? [] : ['status']));
+const isLoading = computed(() => scope.value?.expose.isLoading);
+const transactions = computed(() => scope.value?.expose.data?.items ?? []);
+const total = computed(() => scope.value?.expose.data?.pagination?.total_items ?? 0);
 
-      transactions.value = res.items;
-      emit('loaded', res.pagination.total_items);
-    } catch (error) {
-      handleUnknownError(error);
-    } finally {
-      isLoading.value = false;
-    }
+watch(
+  total,
+  () => {
+    emit('loaded', total.value);
   },
-  { immediate: true }
+  { once: true }
 );
```

Comment:
A more correct and elegant way would be to put it directly into the scope where data is loaded, and use the `wheneverFulfilled` helper. 

```ts
wheneverFulfilled(state, (value) => emit('loaded', value.pagination.total_items))
```

Reply:
Thats great idea but I've just realized that there is no need to fetch transactions and blocks - we can use `blocks`, `txs_rejected` and `txs_accepted` fields when fetching peer status in order to get their total count